Userland/Libraries/LibCore/DirectoryEntry.cpp

```diff
@@ -5,10 +5,36 @@
  */
 
 #include "DirectoryEntry.h"
-#include <dirent.h>
+#ifdef AK_OS_SOLARIS
```

Comment:
Parts of this if-def-soup is not ecessarry in thi file, the `from_stat` method is still a valid option when dirent is available